### PR TITLE
Added syncTranslations script

### PR DESCRIPTION
**Pull request by R3dst0ne**
@MarcinOrlowski asked if I could share the script, I used for #759.
Here it is. Since I didn't spend much time on it, there is definitely room for improvement.

### CODE REVIEW

File: tools/syncTranslations.py [hunk 1]
```diff
@@ -0,0 +1,63 @@
+#!/bin/python3
```
Comment (MarcinOrlowski) [inline tools/syncTranslations.py:1]: Hardcoded paths are problematic. Please use `env` instead, i.e. shebang like 

    #!/usr/bin/env python3

File: tools/syncTranslations.py [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+        file.write(lines)
+    return c
+        
+        
+path = "../src/main/resources/resources/logisim/strings/"
+files = ("analyze","circuit", "data", "draw", "file", "fpga", "gui", "hdl", "proj", "soc", "std", "tools", "util")
+langs = ("de", "el", "es", "fr", "pt", "ru", "it", "nl", "ja", "pl")
+
+for file in files:
+#    print(file)
```
Comment (MarcinOrlowski) [inline tools/syncTranslations.py:56]: can't we have commented out code removed? I understand it's just your debug, correct?

**Review by MarcinOrlowski — CHANGES_REQUESTED** [review]
Could you please also add short note to the source header with a few sentences describing what this script is doing and how to use it (i.e. invocation example etc). Also adding copyright header as we have in `*.java` would probably be good idea.

File: tools/syncTranslations.py (hunk 2 above)
Comment (R3dst0ne) [reply tools/syncTranslations.py]: The purpose of the commented code is to show the number of missing translations.
I now added a variable that controls this output.

**Comment by MarcinOrlowski** [comment]
@R3dst0ne could you please also address this:

> Could you please also add short note to the source header with a few sentences describing what this script is doing and how to use it (i.e. invocation example etc). Also adding copyright header as we have in *.java would probably be good idea.

**Comment by maehne** [comment]
How about adding a "dry-run" mode to the script, which just reports whether a sync is necessary? Then, when we could run it as part of the CI checks. What do you think?

**Comment by MarcinOrlowski** [comment]
I think this could go as separate task as probably we could have more checks (i.e. to see it translation ends with the same punctuation marks as originals (however that would need to have some exceptions as I expect i.e. `jp` to not necessary follow :) etc.

**Comment by R3dst0ne** [comment]
This tool is fragile and only really suited for this one task.
> How about adding a "dry-run" mode to the script, which just reports whether a sync is necessary?

With the current implementation this is not possible because it  only saves the translations in a dictionary. Then the original file(e.g. fr) is overwritten with the keys from the English .properties file and the translations are added with the help of the dictionary. Everything that is commented out in the file (e.g. fr) is ignored.

I agree with @MarcinOrlowski that this should be a separate task that can handle more checks.
For a discussion what kind of tool we need for the checks, we should move to asdf.

**Comment by R3dst0ne** [comment]
@BFH-ktt1
You may have merged this one to early. It is still missing parts @MarcinOrlowski requested, and it is still not clear if and how this script should be used.

**Comment by BFH-ktt1** [comment]
@R3dst0ne : no problem, sorry. But we can still improve on it, as such it is already in the tree.

**Pull request merged**